packages/next/src/bin/next.ts

```diff
@@ -55,27 +64,29 @@ class MyRootCommand extends Command {
               ? ['dev']
               : []
 
-        if (isNotStandard || shouldWarnCommands.includes(cmdName)) {
+        if (isNotStandard || shouldWarnCommands.includes(commandName)) {
           warn(NON_STANDARD_NODE_ENV)
         }
       }
 
-      ;(process.env as any).NODE_ENV = process.env.NODE_ENV || defaultEnv
-      ;(process.env as any).NEXT_RUNTIME = 'nodejs'
+      // Set the default NODE_ENV for the command
+      // @ts-expect-error NODE_ENV is not writable
+      process.env.NODE_ENV = process.env.NODE_ENV || defaultEnv
```

Comment:
The previous code was better since it was more granular with regards to where soundness is broken. `@ts-expect-error` will mask every error on the line.

Reply:
I'd argue casting process.env to any also masks errors. Throughout the codebase process.env isn't cast.

See https://github.com/vercel/next.js/blob/0dfcdea234f2a9f9df29b80037c8d40e4f059223/packages/next/src/server/lib/router-server.ts#L84

Leaving this for know, happy to change it after feedback!

Reply:
> I'd argue casting process.env to any also masks errors.

I didn't argue that it is perfectly sound. It simply masks less legit errors.

Reply:
Very true, maybe an idea to look into some env typesafety or typehints like t3 env does, but more internal?

Reply:
Done, thanks 👍